distant: init at 0.20.0

nyxar77 opened the pull request:
add distant remote development tool, used in also in libraries like distant.nvim


<!--
^ Please summarise the changes you have done and explain why they are necessary here ^

For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [x] aarch64-linux
  - [x] x86_64-darwin
  - [x] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [25.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) (or backporting [24.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) and [25.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#reviewing-contributions
-->

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

nyxar77 committed:
Create default.nix

 add distant remote development tool, used in also in libraries like distant.nvim

mweinelt commented on pkgs/tools/networking/distant/default.nix:
Move to pkgs/by-name/di/distant/package.nix

fe2-Nyxar replied:
first time, thanks for your help

mweinelt commented on pkgs/tools/networking/distant/default.nix:
```diff
@@ -0,0 +1,38 @@
+    description = "Distant - A fast, secure remote development tool";
+    homepage = "https://github.com/chipsenkbeil/distant";
+    license = lib.licenses.mit;
+    maintainers = [
+      "chipsenkbeil"
+      "westernwontons"
+    ];
```
Maintainers are supposed to be handles defined in lib/maintainers.nix.

Check the header of that file and create a separate commit for each maintainer. The package commit should come after.

fe2-Nyxar replied:
i couldn't find the file you are talking about, also these are the maintainers of the project so logically they can be here

mweinelt replied:
The file is at `./maintainers/maintainer-list.nix`, but the references are available at `lib.maintainers`.

fe2-Nyxar replied:
so it should be like add maintainers to the maintainers/maintainer-list.nix
and add this line in default.nix?
```nix
maintainers = [ lib.maintainers.user ]
```
tell me if i'm corret cause i'm confused

mweinelt replied:
Yes, add a maintainer object at the correct location to the file and reference it as `lib.maintainters.chipseenkbeil` if you name the object that.

fe2-Nyxar replied:
thank you so much, big help
i just got done, tell me if i'm missing anything :)

mweinelt reviewed (changes requested):
The commit message is not following the contribution guidelines.

https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#commit-conventions

nyxar77 committed:
Update maintainer-list.nix

nyxar77 committed:
Delete pkgs/tools/networking/distant directory

nyxar77 committed:
distant: init at 0.20.0 

add distant remote development tool, used in also in libraries like distant.nvim

nyxar77 committed:
Update default.nix

nyxar77 committed:
Rename default.nix to package.nix

fe2-Nyxar commented:
@ofborg eval

winterqt commented on maintainers/maintainer-list.nix:
```diff
@@ -3894,6 +3894,12 @@
     keys = [ { fingerprint = "099E 3F97 FA08 3D47 8C75  EBEC E0EB AD78 F019 0BD9"; } ];
   };
+  chipsenkbeil = {
+    email = "nyxar49@gmail.com";
```
Why are these all your email?

Also: these seem to be the maintainers of the upstream project. You should only add yourself as a maintainer in this case.

fe2-Nyxar replied:
thanks for clarification, i've removed them

nyxar77 committed:
Update maintainer-list.nix

nyxar77 committed:
Update package.nix

nyxar77 committed:
Merge branch 'NixOS:master' into master

pluiedev commented on pkgs/by-name/di/distant/package.nix:
```diff
@@ -0,0 +1,37 @@
+
+  src = fetchFromGitHub {
+    owner = "chipsenkbeil";
+    repo = pname;
```
`repo = pname` is an anti-pattern
```suggestion
    repo = "distant";
```

pluiedev commented on pkgs/by-name/di/distant/package.nix:
```diff
@@ -0,0 +1,37 @@
+  src = fetchFromGitHub {
+    owner = "chipsenkbeil";
+    repo = pname;
+    rev = "v${version}";
```
```suggestion
    rev = "refs/tags/v${version}";
```

fe2-Nyxar replied:
I can't squash right now because I have some requested changes

pluiedev commented on pkgs/by-name/di/distant/package.nix:
```diff
@@ -0,0 +1,37 @@
+    perl
+  ];
+  buildInputs = [ openssl ];
+  doCheck = false;
```
Add a comment explaining why tests are disabled (e.g. no tests/failing tests)

pluiedev commented on pkgs/by-name/di/distant/package.nix:
```diff
@@ -0,0 +1,37 @@
+  doCheck = false;
+
+  meta = {
+    description = "Distant - A fast, secure remote development tool";
```
```suggestion
    description = "Fast, secure remote development tool";
```

pluiedev reviewed (changes requested):
Some nitpicks. Also, please squash your commits into just two:
* `maintainers: add fe2-Nyxar`
* `distant: init at 0.20.0`

nyxar77 committed:
Update pkgs/by-name/di/distant/package.nix

Co-authored-by: Leah Amelia Chen <github@acc.pluie.me>

nyxar77 committed:
Update pkgs/by-name/di/distant/package.nix

Co-authored-by: Leah Amelia Chen <github@acc.pluie.me>

nyxar77 committed:
Merge branch 'NixOS:master' into master

nyxar77 committed:
Merge branch 'NixOS:master' into master
